src/xinterpreter_raw.cpp

```diff
@@ -324,4 +322,26 @@ namespace xpyt
         sys.attr("stderr") = stream_module.attr("Stream")("stderr");
     }
 
+
+    std::string raw_interpreter::remove_magics(const std::string& code, bool warning)
+    {
+        std::string result = "";
+        std::stringstream stream(code);
+        std::string temp;
+        while (std::getline(stream, temp)) {
+            if (temp.rfind("%", 0) != 0) {
```

Comment:
I believe this doesn't work. For example the following code cell will not work though it's valid Python:

```python
a = ("hey %x"
% 3)
```

This fails with a SyntaxError because the cleaned code ends up being:
```python
a = ("hey %x"
import logging
logging.getLogger().warning('IPython magics are disabled in raw mode')
```
which is not valid

Reply:
I guess you want to check only the first non-empty line of code for magics?

Is it valid to have a magic in the middle of the code anyway?

Reply:
Indeed my logic does not work well, it'll fix it, thanks @martinRenou

Reply:
Mmh that's annoying. I wonder if we could strip down the IPython's input transformer and reuse it a bit?

Reply:
Or it might be better to just let the execution fail with a `SyntaxError`

Reply:
```
---------------------------------------------------------------------------
SyntaxError                               Traceback (most recent call last)
File /home/trungle/miniconda3/envs/xpydev/lib/python3.9/ast.py, in parse:
Line 50:    return compile(source, filename, mode, flags,

SyntaxError: invalid syntax (<string>, line 2)
---------------------------------------------------------------------------
There may be Ipython magics in your code, this feature is not supported in xeus-python raw mode!
```

@martinRenou  something like this will do the job ? it will add a  warning on `SyntaxError` with `%` symbol in code.

Reply:
👍🏽 Yep

We can also suggest the user to either use classic xeus-python or get rid of the magic